contracts/EarthfastProjects.sol

```diff
@@ -234,28 +234,36 @@ contract EarthfastProjects is AccessControlUpgradeable, PausableUpgradeable, Ree
   }
 
   /// @notice Transfers USDC into the contract and applies them toward given operator stake.
-  /// @dev Needs either a token allowance from msg.sender, or a gasless approval (v/r/s != 0).
+  /// @dev Needs either a token allowance from projectFunder, or a gasless approval (v/r/s != 0).
+  /// This is used instead of explicit input checks to enable third party funding and calling by an Entrypoint proxy contract.
   /// @dev CAUTION: To avoid loss of funds, do NOT deposit to this contract by token.transfer().
-  function depositProjectEscrow(bytes32 projectId, uint256 amount, uint256 deadline, uint8 v, bytes32 r, bytes32 s)
+  /// @param projectFunder The owner of the project.
```

Comment:
is this actually the owner or just the funder sending funds on behalf of owner, regardless of if it's the same wallet

Reply:
It's either the project owner or the third party funding the escrow. I'll update the comment accordingly